include/tallies/tallies.hpp

```diff
@@ -64,6 +66,16 @@ class Tallies {
 
   static Tallies& instance();
 
+  void add_ITally(std::shared_ptr<ITally> new_tally) {
+    new_tally->set_net_weight(total_weight);
+
+    if (new_tally->estimator_str() == "collision") {
+      new_itally_collision_.push_back(new_tally);
+    } else if (new_tally->estimator_str() == "track-length") {
+      new_itally_track_length_.push_back(new_tally);
```

Comment:
Lets change these to not get the string, but the Quantity enum directly. That is a much more efficient comparison.